gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/webauthn/WebAuthnRegisterSuccessEndpoint.java

```diff
@@ -149,19 +153,36 @@ public String getTemplateSuffix() {
 
     private static String getClientOS(HttpServerRequest request) {
         final String browserDetails = Optional.ofNullable(request.getHeader(HttpHeaders.USER_AGENT)).orElse("Unknown");
-        final String lowerCaseBrowser = browserDetails.toLowerCase();
-        if (lowerCaseBrowser.contains("windows")) {
-            return "Windows";
-        } else if (lowerCaseBrowser.contains("mac")) {
-            return "Mac";
-        } else if (lowerCaseBrowser.contains("x11")) {
-            return "Unix";
-        } else if (lowerCaseBrowser.contains("android")) {
-            return "Android";
-        } else if (lowerCaseBrowser.contains("iphone")) {
-            return "IPhone";
-        } else {
-            return "Device";
+        return DeviceBrowser.getFromBrowser(browserDetails).capitalize();
+    }
+
+    public enum DeviceBrowser {
+        ANDROID("android"),
+        IPHONE("iphone"),
+        MAC("mac"),
+        UNIX("x11"),
+        WINDOWS("windows"),
+        DEVICE("device");
+
+        private final String device;
+
+        DeviceBrowser(String device) {
+            this.device = device;
+        }
+
+        public static DeviceBrowser getFromBrowser(String userAgent) {
```

Comment:
Renamed the method to reflect the fact we want to get a `DeviceBrowser` from a User Agent